libs/langchain/langchain/vectorstores/pinecone.py

```diff
@@ -303,6 +325,44 @@ def max_marginal_relevance_search(
             embedding, k, fetch_k, lambda_mult, filter, namespace
         )
 
+    @classmethod
+    def get_pinecone(
+        cls,
+        index_name: Optional[str],
+        embedding: Union[Embeddings, Callable],
+        text_key: str,
+        namespace: Optional[str] = None,
+        pool_threads: int = 1,
+        distance_strategy: Optional[DistanceStrategy] = DistanceStrategy.COSINE,
+        **kwargs: Any,
+    ) -> Pinecone:
+        """Get or create a Pinecone index."""
+
+        try:
+            import pinecone
+        except ImportError:
+            raise ValueError(
+                "Could not import pinecone python package. "
+                "Please install it with `pip install pinecone-client`."
+            )
+
+        indexes = pinecone.list_indexes()  # checks if provided index exists
+
+        if index_name in indexes:
+            index = pinecone.Index(index_name, pool_threads=pool_threads)
+        elif len(indexes) == 0:
+            raise ValueError(
+                "No active indexes found in your Pinecone project, "
+                "are you sure you're using the right API key and environment?"
```

Comment:
```suggestion
                "are you sure you're using the right Pinecone API key and Environment? Please double check your Pinecone dashboard."
```

Reply:
Don't feel like you have to just accept my suggestions - Let me know (LMK) if this is too long.